src/pybamm/solvers/processed_variable.py

```diff
@@ -76,45 +72,416 @@ def __init__(
         self.base_eval_size = self.base_variables[0].size
 
         # xr_data_array is initialized
-        self._xr_data_array = None
+        self._xr_array_raw = None
+        self._entries_raw = None
+        self._entries_for_interp_raw = None
+        self._coords_raw = None
+
+    def initialise(self):
+        if self.entries_raw_initialized:
```

Comment:
Why do we need to set these variables that are not used? Be simpler just to return if already initialised?

Reply:
Thanks, I forgot to remove this from a previous version of the function